<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1180</width>
    <height>675</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>900</width>
    <height>515</height>
   </size>
  </property>
  <property name="windowTitle">
   <string notr="true">FreeJoy QtConfigurator</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="layoutG_MainLayout" rowstretch="10,1" columnstretch="1,5">
    <property name="leftMargin">
     <number>6</number>
    </property>
    <property name="topMargin">
     <number>6</number>
    </property>
    <property name="rightMargin">
     <number>6</number>
    </property>
    <property name="bottomMargin">
     <number>6</number>
    </property>
    <item row="1" column="0" colspan="2">
     <layout class="QVBoxLayout" name="layoutV_DebugWindow">
      <property name="spacing">
       <number>0</number>
      </property>
     </layout>
    </item>
    <item row="0" column="1">
     <widget class="QTabWidget" name="tabWidget">
      <property name="baseSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab_PinConfig">
       <attribute name="title">
        <string> Pin Config </string>
       </attribute>
       <layout class="QVBoxLayout" name="layoutV_tabPinConfig">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>6</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_ButtonConfig">
       <attribute name="title">
        <string> Button Config </string>
       </attribute>
       <layout class="QVBoxLayout" name="layoutV_tabButtonConfig">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>6</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_AxesConfig">
       <attribute name="title">
        <string> Axes Config </string>
       </attribute>
       <layout class="QVBoxLayout" name="layoutV_tabAxesConfig">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>6</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_AxesCurves">
       <attribute name="title">
        <string> Axes Curves </string>
       </attribute>
       <layout class="QVBoxLayout" name="layoutV_tabAxesCurvesConfig">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>6</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_ShiftRegisters">
       <attribute name="title">
        <string> Shift Registers </string>
       </attribute>
       <layout class="QVBoxLayout" name="layoutV_tabShiftRegistersConfig">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>6</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_Encoders">
       <attribute name="title">
        <string> Encoders</string>
       </attribute>
       <layout class="QVBoxLayout" name="layoutV_tabEncodersConfig">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>6</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_LED">
       <attribute name="title">
        <string> LED/PWM </string>
       </attribute>
       <layout class="QVBoxLayout" name="layoutV_tabLedConfig">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>6</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_AdvancedSettings">
       <attribute name="title">
        <string> Advanced Settings </string>
       </attribute>
       <layout class="QVBoxLayout" name="layoutV_tabAdvSettings">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>6</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_Debug">
       <attribute name="title">
        <string> Debug </string>
       </attribute>
       <widget class="QPushButton" name="pushButton_TestButton">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>10</y>
          <width>71</width>
          <height>31</height>
         </rect>
        </property>
        <property name="text">
         <string notr="true">1</string>
        </property>
       </widget>
       <widget class="QPushButton" name="pushButton_TestButton_2">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>50</y>
          <width>71</width>
          <height>31</height>
         </rect>
        </property>
        <property name="text">
         <string>2</string>
        </property>
       </widget>
      </widget>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QGroupBox" name="groupBox_4">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="title">
       <string>Device</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_7" rowstretch="0,0,0,0,0,0,0,0,0,0,0,0,0,0,0" columnstretch="0,0">
       <property name="leftMargin">
        <number>6</number>
       </property>
       <property name="topMargin">
        <number>6</number>
       </property>
       <property name="rightMargin">
        <number>6</number>
       </property>
       <property name="bottomMargin">
        <number>6</number>
       </property>
       <property name="horizontalSpacing">
        <number>6</number>
       </property>
       <property name="verticalSpacing">
        <number>9</number>
       </property>
       <item row="9" column="0" colspan="2">
        <widget class="QPushButton" name="pushButton_LoadFromFile">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>165</width>
           <height>37</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">height: 30px;</string>
         </property>
         <property name="text">
          <string>Load config from file</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0" colspan="2">
        <widget class="QLabel" name="label_DeviceStatus">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>20</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true"/>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="8" column="1">
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="11" column="0" colspan="2">
        <widget class="QPushButton" name="pushButton_SaveToFile">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>165</width>
           <height>37</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">height: 30px;</string>
         </property>
         <property name="text">
          <string>Save config to file</string>
         </property>
        </widget>
       </item>
       <item row="13" column="0" colspan="2">
        <widget class="QPushButton" name="pushButton_ResetAllPins">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>165</width>
           <height>37</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">height: 30px;</string>
         </property>
         <property name="text">
          <string>Reset All</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0" colspan="2">
        <widget class="CenteredCBox" name="comboBox_HidDeviceList">
         <property name="minimumSize">
          <size>
           <width>165</width>
           <height>26</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">height: 22px;</string>
         </property>
        </widget>
       </item>
       <item row="12" column="1">
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="5" column="1">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="2" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="1,1,1">
         <property name="spacing">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <item>
          <spacer name="horizontalSpacer_9">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Preferred</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_Wiki">
           <property name="cursor">
            <cursorShape>PointingHandCursor</cursorShape>
           </property>
           <property name="styleSheet">
            <string notr="true"/>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="icon">
            <iconset resource="resources.qrc">
             <normaloff>:/Images/ST_wiki.png</normaloff>:/Images/ST_wiki.png</iconset>
           </property>
           <property name="iconSize">
            <size>
             <width>30</width>
             <height>30</height>
            </size>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_6">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Preferred</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item row="6" column="0" colspan="2">
        <widget class="QPushButton" name="pushButton_LoadDefaultConfig">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>165</width>
           <height>37</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">height: 30px;</string>
         </property>
         <property name="text">
          <string>Load default config</string>
         </property>
        </widget>
       </item>
       <item row="14" column="0" colspan="2">
        <widget class="QPushButton" name="pushButton_ShowDebug">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>165</width>
           <height>37</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">height: 30px;</string>
         </property>
         <property name="text">
          <string>Show debug</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0" colspan="2">
        <widget class="QPushButton" name="pushButton_WriteConfig">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>165</width>
           <height>37</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">height: 30px;</string>
         </property>
         <property name="text">
          <string>Write config to Device</string>
         </property>
        </widget>
       </item>
       <item row="7" column="0" colspan="2">
        <widget class="QPushButton" name="pushButton_SetDefaultConfig">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>165</width>
           <height>37</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">height: 30px;</string>
         </property>
         <property name="text">
          <string>Set file to default config</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0" colspan="2">
        <widget class="QPushButton" name="pushButton_ReadConfig">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>165</width>
           <height>37</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">height: 30px;</string>
         </property>
         <property name="text">
          <string>Read config from Device</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CenteredCBox</class>
   <extends>QComboBox</extends>
   <header>centered_cbox.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections/>
</ui>
